### DISCUSSION TITLE
Enable/Disable rule

### DISCUSSION THREAD

**Discussion by chkhalafov** [PaloAltoNetworks/pan-os-python#458] [Q&A] [answered]
Dear developers

How I can enable/disable the rule without affecting other parameters of the rule? 
I tried: security_rule = SecurityRule("Disaster-Allow", disabled=False)

But in this case, "pandevice" requires an action parameter. 
security_rule = SecurityRule("Disaster-Allow", disabled=False, action="allow") this works but, if not define source/destination zones/addresses it add "any" as source/destination zone/address. It is not comfortable to define all parameters of the rule.

Is there any method to solve this?

**Comment by shinmog** [comment, marked as answer]
Few different ways to do this.

If you have multiple params you want to change about an object, you can `.apply()` it.  But if you just want to update a single param, you can use [update](https://pan-os-python.readthedocs.io/en/latest/module-base.html#panos.base.PanObject.update):

```python
from panos.firewall import Firewall
from panos.policies import Rulebase, SecurityRule

fw = Firewall(.........)

base = Rulebase()
fw.add(base)

rule = SecurityRule("rule name here", disabled=True)
base.add(rule)

rule.update("disabled")
```

**Reply by chkhalafov** [reply]
@shinmog 
Thank you. It is useful. But what about removing the virtual router under the interface.
I've tried:

```python
vr = network.VirtualRouter(name="default")
fw.add(vr)

vr.update("interface=ethernet1/1.10")
```

It returns an error. The script removes all interfaces under the default virtual router if the interface name is not defined.

Also, I tried:

```python
my_interface = network.Interface("ethernet1/1.10")
fw.add(my_interface)
my_interface.set_virtual_router("default", refresh=True)
my_interface.update("default-router")

```
But there is no "default-router" keyword.

Is there any method to get all keywords for each object?
